File: 25/d3/second.cc
#include <iostream>
#include <fstream>
#include <string>
#include <cmath>

int main() {
    long long sum = 0;
    int status = 1;
    std::ifstream f("list");
    std::string line;
    while (std::getline(f, line)) {
        std::cout << status << "\n";
        status++;
        long long maxv = 0;
        int n = line.size();
        for (int a = 0; a < n; a++)
            for (int b = a + 1; b < n; b++)
                for (int c = b + 1; c < n; c++)
                    for (int d = c + 1; d < n; d++)
                        for (int e = d + 1; e < n; e++)
                            for (int f2 = e + 1; f2 < n; f2++)
                                for (int g = f2 + 1; g < n; g++)
                                    for (int h = g + 1; h < n; h++)
                                        for (int i = h + 1; i < n; i++)
                                            for (int j = i + 1; j < n; j++)
                                                for (int k = j + 1; k < n; k++)
                                                    for (int l = k + 1; l < n; l++) {
                                                        long long val =
                                                            (line[a] - '0') * (long long)std::pow(10, 11) +
                                                            (line[b] - '0') * (long long)std::pow(10, 10) +
                                                            (line[c] - '0') * (long long)std::pow(10, 9)  +
                                                            (line[d] - '0') * (long long)std::pow(10, 8)  +
                                                            (line[e] - '0') * (long long)std::pow(10, 7)  +
                                                            (line[f2] - '0') * (long long)std::pow(10, 6) +
                                                            (line[g] - '0') * (long long)std::pow(10, 5)  +
                                                            (line[h] - '0') * (long long)std::pow(10, 4)  +
                                                            (line[i] - '0') * (long long)std::pow(10, 3)  +
                                                            (line[j] - '0') * (long long)std::pow(10, 2)  +
                                                            (line[k] - '0') * 10LL +
                                                            (line[l] - '0');

                                                        if (val > maxv)
                                                            maxv = val;
                                                        }

                                                        sum += maxv;
                                                    }
    std::cout << sum << "\n";
    return 0;
}

File: 25/d3/second.py
sum = 0
status = 1
with open("list", "r") as f:
    for line in f:
        print(status)
        line = line.strip()
        tiotal, num = 0, 0
        if not line:
            continue
        #WHY? because i can.
        max = 0
        for a in range(0 , len(line)):
            for b in range(a + 1 , len(line)):
                for c in range(b + 1, len(line)):
                    for d in range(c + 1, len(line)):
                        for e in range(d + 1, len(line)):
                            for f in range(e + 1, len(line)):
                                for g in range(f + 1, len(line)):
                                    for h in range(g + 1, len(line)):
                                        for i in range(h + 1, len(line)):
                                            for j in range(i + 1, len(line)):
                                                for k in range(j + 1, len(line)):
                                                    for l in range(k + 1, len(line)):
                                                        if max < int(line[a]) * pow(10, 11) + int(line[b]) * pow(10, 10) + int(line[c]) * pow(10, 9) + int(line[d]) * pow(10, 8) + int(line[e]) * pow(10, 7) +int(line[f]) * pow(10, 6) + int(line[g]) * pow(10, 5) + int(line[h]) * pow(10, 4) + int(line[i]) * pow(10, 3) +  int(line[j]) * pow(10, 2) + int(line[k]) * 10 + int(line[l]):
                                                            max = int(line[a]) * pow(10, 11) + int(line[b]) * pow(10, 10) + int(line[c]) * pow(10, 9) + int(line[d]) * pow(10, 8) + int(line[e]) * pow(10, 7) +int(line[f]) * pow(10, 6) + int(line[g]) * pow(10, 5) + int(line[h]) * pow(10, 4) + int(line[i]) * pow(10, 3) +  int(line[j]) * pow(10, 2) + int(line[k]) * 10 + int(line[l]) 


        sum += max
print(sum)
